cleverpush/src/main/java/com/cleverpush/banner/AppBannerCarouselAdapter.java

```diff
@@ -335,6 +350,34 @@ private void composeImageBlock(LinearLayout body, BannerImageBlock block, int po
     }
   }
 
+  /**
+   * Checks whether the content at the specified URL is a GIF image.
+   *
+   * @param imageUrl The URL of the image to be checked.
+   * @return True if the content is a GIF image, false otherwise.
+   */
+  private boolean isGif(String imageUrl) {
```

Comment:
We already load the content of the URL here: https://github.com/cleverpush/cleverpush-android-sdk/pull/180/files#diff-69346692aa9d5de5421d79019f283488c8745bbb0b1ece567f6746d695fd34bfR318

We should not do it again. Please work with the existing inputStream or bitmap